packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2157,19 +2263,29 @@ abstract class Element implements BuildContext {
     return renderObjectAncestor?.renderObject;
   }
 
-  /// Calls visitor for each ancestor element.
-  ///
-  /// Continues until visitor reaches the root or until visitor returns false.
   @override
   void visitAncestorElements(bool visitor(Element element)) {
     Element ancestor = _parent;
     while (ancestor != null && visitor(ancestor))
       ancestor = ancestor._parent;
   }
 
+  /// Called when a dependencies of this element changes.
+  ///
+  /// The [inheritFromWidgetOfExactType] causes the [InheritedElement]
+  /// associated with the returned [InheritedWidget] to remember this element
+  /// notify this element (by calling this method) whenever its widget is
+  /// updated in a way that causes [InheritedWidget.updateShouldNotify] to
+  /// return true.
+  ///
+  /// In this way, [dependenciesChanged] is a hook that notifies this element
+  /// that the inherited information it depends upon might have changed.
```

Comment:
Rewrote to make it clearer that the element updating is just one possible way for this function to get called.